src/mindlessgen/molecules/generate_molecule.py

```diff
@@ -431,3 +459,70 @@ def check_distances(
             if r < scale_minimal_distance * sum_radii:
                 return False
     return True
+
+
+def fixed_charge_correction(
+    cfg: GenerateConfig,
+    natoms: np.ndarray,
+    nel: int,
+    valid_elems: set[int],
+    verbosity: int,
+) -> np.ndarray:
+    """
+    Correct the number of electrons if a fixed charge is given and the molecule is not closed shell.
+    """
+    f_elem = any(
+        count > 0 and (i in get_lanthanides() or i in get_actinides())
+        for i, count in enumerate(natoms)
+    )
+    if f_elem:
+        ligand_protons = calculate_ligand_protons(natoms, nel)
+        # If f block elements are included, correct only if the remaning ligand protons are uneven
+        if ligand_protons % 2 != 0:
+            natoms = fixed_charge_elem_correction(cfg, natoms, valid_elems, verbosity)
+            return natoms
+    # If f block elements are not included, correct if the number of electrons is uneven
+    elif nel % 2 != 0 and f_elem is False:
```

Comment:
```suggestion
    elif nel % 2 != 0:
```

With `elif`, `f_elem` has to `False` anyway.